documentation/boltspec_reference.md

```diff
@@ -4,6 +4,180 @@ The `BoltSpec` helper library includes several functions to help write unit
 tests for plans. For more information about writing unit tests for plans, and
 how to use these functions, see [Testing plans](testing_plans.md).
 
+## Any stubs
+
+Plans often execute sub-plans with the `run_plan` function to build complex
+workflows. When testing these plans, it might be helpful to execute any
+sub-plans as well without needing to stub or mock the plan. To support this,
+`BoltSpec` offers two different modes:
```

Comment:
This intro paragraph should more closely match the intro for "Stubs". Because the `allow_any_*` stubs allow you to stub _all_ related function invocations, we should mention where they might be helpful (such as when you invoke `run_task` many times in a plan and don't want to invoke `allow_task` for each individual task).